backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -1295,10 +1295,50 @@ private boolean checkAllResourceAllowed(List<Resources> resources) {
   }
 
   public InternalFuture<ExperimentRun> createExperimentRun(CreateExperimentRun request) {
-    return getEntityPermissionBasedOnResourceTypes(
-            Collections.singletonList(request.getProjectId()),
-            ModelDBActionEnum.ModelDBServiceActions.UPDATE,
-            ModelDBServiceResourceTypes.PROJECT)
+    // Validate arguments
+    var futureTask =
+        InternalFuture.runAsync(
+            () -> {
+              if (request.getProjectId().isEmpty()) {
+                throw new InvalidArgumentException(
+                    "Project ID not found in CreateExperimentRun request");
+              } else if (request.getExperimentId().isEmpty()) {
+                throw new InvalidArgumentException(
+                    "Experiment ID not found in CreateExperimentRun request");
+              }
+            },
+            executor);
+    return futureTask
+        .thenCompose(
+            unused ->
+                getEntityPermissionBasedOnResourceTypes(
+                    Collections.singletonList(request.getProjectId()),
+                    ModelDBActionEnum.ModelDBServiceActions.UPDATE,
+                    ModelDBServiceResourceTypes.PROJECT),
+            executor)
+        .thenAccept(
+            allowed -> {
+              if (!allowed) {
+                throw new PermissionDeniedException("Permission denied");
+              }
+            },
+            executor)
+        .thenCompose(
+            unused ->
+                jdbi.useHandle(
+                    handle -> {
+                      Long count =
+                          handle
+                              .createQuery("SELECT COUNT(*) FROM experiment where id = :id")
+                              .bind("id", request.getExperimentId())
+                              .mapTo(Long.class)
+                              .one();
+                      if (count == 0) {
+                        throw new NotFoundException(
+                            "Experiment not found for given ID: " + request.getExperimentId());
+                      }
+                    }),
```

Comment:
Experiment exists or not is missing before creating ER so i have added it here and due to this change i have to moved requested parameters validation from Create ER handler to above.